### DISCUSSION TITLE
Add token.like_roman_num

### DISCUSSION THREAD

**Discussion by filipecaixeta** [explosion/spaCy#6217] [New Features & Project Ideas]
## Feature description
Add a feature to the library to identify roman numerals (I, II, III, IV, V, VI, VII, VIII, IX, X, XI, XII, ...)

I want to add `token.like_roman_num` to return a boolean if it is roman numeral and `token.roman_num_value` to convert it to integer

It it is ok I would like to work on this feature.

**Comment by adrianeboyd** [comment]
This is a nice idea, but it's specific enough that I'm not sure it makes sense to incorporate it as a feature in the core library.

For your own implementation, you could implement both as custom token extensions (https://spacy.io/usage/processing-pipelines#custom-components-attributes), or alternatively, `like_roman_num` could be implemented as a flag: https://spacy.io/api/vocab#add_flag . The main advantage of the flag is that it would be slightly faster because it would store the flag value in the vocab for any roman numerals that have been seen before, but in practice it probably doesn't make a huge difference and having both defined in the same way as custom extensions might be simpler to maintain.

**Comment by danijel3** [comment]
Just in case someone else is looking:

```
def is_roman_numeral(text: str) -> bool:
    for c in text:
        if c not in {'I', 'V', 'X', 'L', 'D', 'C', 'M'}:
            return False
    return True


ROMAN_NUM = nlp.vocab.add_flag(is_roman_numeral)

for tok in nlp('...'):
    if tok.check_flag(ROMAN_NUM):
        print(tok)
```
